### ISSUE TITLE
Some animations don't get disabled when browser's reduced motion setting is enabled

### ISSUE THREAD

**Issue by GCHQ-Developer-847** [mi6/ic-ui-kit#4059]
### Summary of the bug 
When `prefers-reduced-motion: reduce` is set, all animations / transitions should be disabled as per the [2.3.3 Animation from Interactions](https://www.w3.org/TR/WCAG21/#animation-from-interactions) WCAG requirement, but they still happen on some components.

### 🪜 How to reproduce 
For example, on the Accordion component:
1. On Chrome, go to the [Accordion page](https://design.sis.gov.uk/components/accordion/code).
2. Open the Chrome DevTools and go to the Rendering tab (press Cmd+Shift+P or Ctrl+Shift+P, and type "Rendering").
3. Scroll to "Emulate CSS media feature prefers-reduced-motion" and select the "reduce" option.
4. Open and close the accordion.
5. See that the sliding up and down animation still happens (it should just immediately open / close).

This also happens on other components:
- Side navigation sliding in and out
- Toast slides / fades in and out

There are also a few more instances of this but not sure if we want / need to do anything about these:
- Focus indicator on most components fades in and out
- Loading indicator keeps moving
- Skeleton keeps flashing

I think there might be more components I've missed so it would be worth checking others (maybe search for `transition` in CSS files).

### 🧐 Expected behaviour 
Animations should be disabled. E.g. the accordion should immediately switch to the open / close state with no transition.

### 🚨 Urgency (low, medium or high)
I think this is probably low urgency because it's a Level AAA requirement (so not essential for WCAG conformance).
